push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/SenderServiceImpl.java

```diff
@@ -67,6 +67,8 @@ public void send(PushApplication pushApplication, UnifiedPushMessage message) {
                         message.getClientIdentifier()
                         );
 
+        //Let's put a unique identifier into the payload to be able to do some analytics around that Push Notification
+        message.getMessage().getUserData().put("push-identifier",pushMessageInformation.getId());
```

Comment:
+1 should be nested on a different spot of the message, outside of the actual payload content